clcache.py

```diff
@@ -164,33 +164,43 @@ def clean(self, stats, maximumSize):
         except ImportError:
             walker = os.walk
 
-        objects = [os.path.join(root, "object")
-                   for root, folder, files in walker(self.objectsDir)
-                   if "object" in files]
+        objectInfos = []
+        for x in xrange(16):
+            try:
+                objects = [os.path.join(root, "object")
+                       for root, folder, files in walker(self.objectsDir)
+                       if "object" in files]
+
+                objectInfos = [(os.stat(fn), fn) for fn in objects]
```

Comment:
Sure they will and I am a heavy abuser of this myself (since there was no -C for a long time). But when I do so, I wait until my compilations are done and otherwise live with build errors. When you're deleting the cache while building, you risk restoring corrupted object files without noticing it, which is something you should avoid.

The patch does not add runtime overhead, right. But it adds code complexity and WTF moments when trying to understand the code. This can be done, but I don't see that it is worth it here.